waspc/data/Generator/templates/server/src/webSocket.ts

```diff
@@ -0,0 +1,80 @@
+{{={= =}=}}
+
+import http from 'http'
+import { Server, Socket } from 'socket.io'
+import { EventsMap, DefaultEventsMap } from '@socket.io/component-emitter'
+
+import config from './config.js'
+import prisma from './dbClient.js'
+
+{=# isAuthEnabled =}
+import { getUserFromToken } from './core/auth.js'
+{=/ isAuthEnabled =}
+
+{=& userWebSocketFn.importStatement =}
+
+export type WebSocketDefinition<
+  ClientToServerEvents extends EventsMap = DefaultEventsMap,
+  ServerToClientEvents extends EventsMap = DefaultEventsMap,
+  InterServerEvents extends EventsMap = DefaultEventsMap,
+  SocketData extends WaspSocketData = WaspSocketData
+> =
+  ( io: Server<ClientToServerEvents, ServerToClientEvents, InterServerEvents, SocketData>,
+    context: {
+      entities: {
+        {=# allEntities =}
+        {= name =}: typeof prisma.{= prismaIdentifier =},
+        {=/ allEntities =}
+      }
+    }
+  ) => Promise<void> | void
+
+export interface WaspSocketData {
+  user?: any;
+}
+
+type WebSocketFn = typeof {= userWebSocketFn.importIdentifier =}
+type ServerType = Parameters<WebSocketFn>[0]
+type Events = ServerType extends Server<infer ClientToServerEvents, infer ServerToClientEvents>
+  ? [ClientToServerEvents, ServerToClientEvents]
+  : [DefaultEventsMap, DefaultEventsMap]
```

Comment:
How come we're handling this case? Won't the code break (in runtime) if those types aren't correct anyway?

Reply:
I was thinking of users that don't use `WebSocketDefinition` in their definition or use Javascript. 

Let me test it a bit more and see if it's needed 👍

Reply:
I just realised I needed this check anyways to `infer` the `ClientToServerEvents` and `ServerToClientEvents` types. And in the case it doesn't infer, just return the default ones. Does this make sense?

Reply:
![Screenshot 2023-06-16 at 12 39 43](https://github.com/wasp-lang/wasp/assets/2223680/e67026eb-c4dc-48c1-9461-92239be7ce1c)
![Screenshot 2023-06-16 at 12 39 47](https://github.com/wasp-lang/wasp/assets/2223680/a494eb02-3fe5-422e-81b6-4c5eed5af4bd)
![Screenshot 2023-06-16 at 12 40 00](https://github.com/wasp-lang/wasp/assets/2223680/bf963ee7-dfeb-4fe6-803c-df9875603865)